### ISSUE TITLE
[Terraform] Lambda: Unable to find handler script in local mount in non-remote Docker

### ISSUE THREAD

**Issue by bmitmanski** [localstack/localstack#2579]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

There is an error using Terraform to initialize multiple Lambdas. According to code, the exception shouldn't be raised:
https://github.com/localstack/localstack/blob/master/localstack/services/awslambda/lambda_api.py#L740

Unfortunately, this error happens occasionally, so it's hard to find a cause. Maybe some kind of race?                                                                                    

## Expected behaviour

Skip checking Lambda archive in local mount in non-remote Docker.

## Actual behaviour

localstack.services.awslambda.lambda_api: Unable to find handler script (LOCAL_PATH) in Lambda archive. Config for local mount, docker, remote: "True", "True", "False"

# Steps to reproduce

## Command used to start LocalStack

```  
localstack:
    image: localstack/localstack
    networks:
      - localstack-network
    ports:
      - 4566:4566
      - 4572:4572
      - 4574:4574
      - 4576:4576
      - 4586:4586
      - 8080:8080
    environment:
      - DEBUG=1
      - HOSTNAME_EXTERNAL=localstack
      - SERVICES=sqs,lambda,cloudwatch,s3,iam,ec2,stepfunctions,cloudwatchlogs
      - LAMBDA_REMOTE_DOCKER=false
      - LAMBDA_EXECUTOR=docker
      - LAMBDA_DOCKER_NETWORK=localstack-network
      - AWS_ACCESS_KEY_ID=123
      - AWS_SECRET_ACCESS_KEY=xyz
      - DEFAULT_REGION=us-east-1
      - DOCKER_HOST=unix:///var/run/docker.sock
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /tmp/localstack:/tmp/localstack
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

Lambda terraform example:
Terraform version: 0.11.14
```
resource "aws_lambda_function" "lambda_function" {
  s3_bucket                       = "__local__"
  s3_key                          = "local/absolute/path"
  function_name                   = "some-function"
  role                            = "somerole"
  handler                         = "index.myHandler"
  runtime                         = "nodejs12.x"
}
```

**Comment by andrei-gypaetus** [comment]
I had the same issue. It was caused by the zip structure. Instead of having 
```
Archive:  my-function.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
      111  2011-11-11 11:11   subfolder/file
      111  2011-11-11 11:11   subfolder/file
999999  2020-06-12 19:47   bootstrap
```
 I had 
```
Archive:  my-function.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
      111  2011-11-11 11:11   my-function/subfolder/file
      111  2011-11-11 11:11  my-function/subfolder/file
999999  2020-06-12 19:47   my-function/bootstrap
```.
You can check that by doing an `unzip -l my-function.zip

**Comment by bmitmanski** [comment]
As I am using local mount I do not provide zip archive, but folder with the lambda source code. 
I have found a workaround by creating lambdas one-by-one instead of parallel terraform creation.

**Comment by whummer** [comment]
Thanks for reporting @bmitmanski @andrei-gypaetus . This should be fixed in #2630 - can you please give it another try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks
